content/integrations/google_cloud/google_cloud_pubsub/Java.md

```diff
@@ -0,0 +1,429 @@
+    consumer.ack();
+    OpenCensusTraceUtil.addAnnotation("Receiver:Done: " + data);
+ }
+}
+
+/** Receive messages over a subscription. */
+public static void main(String... args) throws Exception {
+  // set subscriber id, eg. my-sub
+  String subscriptionId = args[0];
+  ProjectSubscriptionName subscriptionName =
+    ProjectSubscriptionName.of(PROJECT_ID, subscriptionId);
+  Subscriber subscriber = null;
+  try {
+    // create a subscriber bound to the asynchronous
+    // OpenCensus message receiver
+    MessageReceiver receiver =
+      new OpenCensusUtil.OpenCensusMessageReceiver(
+        new MessageReceiverExample());
+    subscriber =
+      Subscriber.newBuilder(subscriptionName, receiver).build();
+    subscriber.startAsync().awaitRunning();
+    // Continue to listen to messages
+    while (true) {
+      Thread.sleep(Long.MAX_VALUE);
+    }
+  } finally {
+    if (subscriber != null) {
+        subscriber.stopAsync();
+    }
+  }
+}
+```
+
+Once again, only one change is really necessary to the Subscriber code, the
+addition of:
+
+```java
+new OpenCensusUtil.OpenCensusMessageReceiver(...)
+```
+
+As previously described, the `OpenCensusMessageReceiver` will wrap the given
+`MessageReceiverExample` so that it is executed in a new root span with the
+propated publisher span set as a parent link.
+
+In our example code, the user-defined `MessageReceiverExample` has been modified
+in three places to add annotations to give insight into what's going in the
+code. Firstly, at the beginning:
+
+```java
+OpenCensusTraceUtil.addAnnotation("Receiver:Message");
+```
+
+Secondly, after processing the message and just before acknowledging it:
+
+```java
+OpenCensusTraceUtil.addAnnotation("Receiver:Ack: " + data);
+```
+
+and, finally, at the very end:
+
+```java
+OpenCensusTraceUtil.addAnnotation("Receiver:Done: " + data);
+```
+
+Note that, for this example, the data sent from the publisher is simple the
+string “message-i” where ‘i’ denotes the i-th message sent and corresponds to
+the publisher’s “PublisherRoot-i” span.
+
+## Running the Example
+The example code provides a Publisher and a Subscriber. Following the original
+example code, we can execute the Publisher specifying how many messages to
+publish. In the following instance, we have the publisher publish 4 messages.
+
+### Publisher Trace
+For the Publisher, we can see the parent span `Publisher` with `Publisher:Begin`
+annotation and 4 child spans (`PublisherRoot-0` ... `PublisherRoot-3`). Each
+child span has their span id added as an annotation. For `PublisherRoot-0`,
+which sends `message-0`, the span id is `98d1b951adaeb95a`:
+
+![publisher-trace](images/publisher-trace.png)
+
+### Subscriber Trace (message-0)
+For the Subscriber, if we look at the span for `message-0`:
+
+![subscriber-trace](images/subscriber-trace.png)
+
+we can see that the associated parent span link is also `98d1b951adaeb95a`:
+
+![subscriber-trace-link](images/subscriber-trace-link.png)
```

Comment:
These need absolute URLs so please place all the images under /static/images and then fix these URLs to be absolute i.e. `/images/subscriber-trace-link.png`

Reply:
Done but the images don't render - is this a real issue or a limitation of the review system?

Reply:
The images don't render because you physically placed them under `content/integrations/google_cloud/google_cloud_pubsub/images/` but they should be physically placed under `static/images` as I had suggested in https://github.com/census-instrumentation/opencensus-website/pull/670#discussion_r275812390